### PR TITLE
feat: add GM support

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Description

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/zh/latest/plugins/gm.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## 描述
+
+`gm` 插件能启用国密相关的功能。目前支持通过该插件动态配置国密双证书。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/gm.md:31]:
```suggestion
`gm` 插件能启用国密相关的功能。目前支持通过该插件动态配置国密双证书。

:::note 相关介绍
国密就是国产化的密码算法。在我们日常开发过程中会接触到各种各样的密码算法，如 RSA、SHA256 等等。为了达到更高的安全等级，许多大公司和国家会制定自己的密码算法。国密就是这样一组由中国国家密码管理局制定的密码算法。在国际形势越发复杂多变的今天，密码算法的国产化替代，在一些领域已经成为了一股势不可挡的潮流。
:::
```

**Review by soulbird — APPROVED** [review]

File: docs/zh/latest/plugins/gm.md (hunk 1 above)
Comment (SylviaBABY) [reply docs/zh/latest/plugins/gm.md:31]: I think it's worth briefly introducing the definition at begining

File: docs/zh/latest/plugins/gm.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+print(resp.text)
+```
+
+将上面的脚本保存为 `./create_gm_ssl.py`，运行：
+
+```shell
+./create_gm_ssl.py t/certs/server_enc.crt  t/certs/server_enc.key t/certs/server_sign.crt t/certs/server_sign.key
+```
+
+输出结果：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/gm.md:162]:
```suggestion
输出结果如下所示：
```

File: docs/zh/latest/plugins/gm.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+```
+...
+New, NTLSv1.1, Cipher is ECDHE-SM2-SM4-CBC-SM3
+...
+```
+
+## 禁用插件
+
+将 `gm` 插件从 `./conf/config.yaml` 移除，然后重启 APISIX 或者通过插件热加载的接口触发插件的卸载。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/gm.md:187]:
```suggestion
如果不再使用此插件，可将 `gm` 插件从 `./conf/config.yaml` 配置文件中移除，然后重启 APISIX 或者通过插件热加载的接口触发插件的卸载。
```

**Review by SylviaBABY — COMMENTED** [review]
LGTM, just modify some parts at the beginning and end

**Review by SylviaBABY — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**